proto-loader: Fix yargs types & update README

badsyntax opened the pull request:
There seems to have been a type update via patch version in `@types/yargs` where `argv._` is now defined as `_: Array<string | number>;`, see:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6de1fd533383aa8fcac9460edd2ee5d36989fa7b/types/yargs/index.d.ts#L643

This is why we need to do `argv._ as string[]` as well as add `parse-positional-numbers: false` parser config to prevent yargs from parsing positional args to numbers. ~I need to use `@ts-ignore` because the types for yargs-parser need updating, see: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50196~

I noticed this as a did a clean clone of this repo follow by a `npm install` and the compile script failed to build.

I also added example usage to the README as I believe this will help with understanding how to use the generator.

badsyntax commented:
@murgatroid99 this is now good for a review.

murgatroid99 commented:
Git sees this as conflicting with the previous PR you sent, so can you resolve that? Also, did you intend to update all of those dependencies? I'm uncertain about the major version bumps to `gts` and especiallly `typescript`.

badsyntax committed:
proto-loader: Add example usage to README

badsyntax committed:
proto-loader: Fix yargs types

badsyntax commented:
I've simplified the changes. 

Not sure why I bumbed `gts` and `typescript` but i've reverted those changes.

And there's no need to bumb any yargs package versions because the [type fixes for `yargs-parser`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50196) are "automatically" pulled in due to the usage of the star (`*`) version number defined in the `@types/yargs` package:

![Screenshot 2021-01-08 at 08 15 29](https://user-images.githubusercontent.com/102141/103990859-c60c5b00-5189-11eb-9728-abaa70d4cd88.png)


I'm sorry for force pushing, but it was easier this way due to the conflicts.
